larpandoracontent/LArObjects/LArMCParticle.h

```diff
@@ -168,15 +249,47 @@ inline pandora::StatusCode LArMCParticleFactory::Write(const Object *const pObje
     if (!pLArMCParticle)
         return pandora::STATUS_CODE_INVALID_PARAMETER;
 
+    const std::vector<pandora::CartesianVector> &mcStepPositions(pLArMCParticle->GetMCStepPositions());
+    const int nMCStepPositions(mcStepPositions.size());
+
+    const std::vector<pandora::CartesianVector> &mcStepMomentas(pLArMCParticle->GetMCStepMomentas());
+    const int nMCStepMomentas(mcStepMomentas.size());
+
     if (pandora::BINARY == fileWriter.GetFileType())
     {
         pandora::BinaryFileWriter &binaryFileWriter(dynamic_cast<pandora::BinaryFileWriter&>(fileWriter));
         PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, binaryFileWriter.WriteVariable(pLArMCParticle->GetNuanceCode()));
+
+        PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, binaryFileWriter.WriteVariable(nMCStepPositions));
+        for (auto const &mcStepPosition : mcStepPositions)
+            PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, binaryFileWriter.WriteVariable(mcStepPosition));
+
+        PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, binaryFileWriter.WriteVariable(nMCStepMomentas));
+        for (auto const &mcStepMomenta : mcStepMomentas)
+            PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, binaryFileWriter.WriteVariable(mcStepMomenta));
     }
     else if (pandora::XML == fileWriter.GetFileType())
     {
         pandora::XmlFileWriter &xmlFileWriter(dynamic_cast<pandora::XmlFileWriter&>(fileWriter));
         PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, xmlFileWriter.WriteVariable("NuanceCode", pLArMCParticle->GetNuanceCode()));
+
+        PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, xmlFileWriter.WriteVariable("NumberOfMCStepPositions", nMCStepPositions));
+        for (int step = 0; step < nMCStepPositions; ++step)
+        {
+            const pandora::CartesianVector &position(mcStepPositions[step]);
+            std::stringstream mcStepPositionName;
+            mcStepPositionName << "MCStepPosition" << step;
+            PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, xmlFileWriter.WriteVariable(mcStepPositionName.str(), position));
+        }
+
+        PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, xmlFileWriter.WriteVariable("NumberOfMCStepMomentas", nMCStepMomentas));
```

Comment:
Prefer `NumberOfMCStepMomenta` here and throughout all related usage. Apologies if you have gigabytes of XML files with this name persisted within!